net/dispatcher.go

```diff
@@ -19,44 +23,158 @@ import (
 	"github.com/frain-dev/convoy/util"
 )
 
+var (
+	ErrAllowListIsRequired = errors.New("allowlist is required")
+	ErrBlockListIsRequired = errors.New("blocklist is required")
+	ErrLoggerIsRequired    = errors.New("logger is required")
+	ErrInvalidIPPrefix     = errors.New("invalid IP prefix")
+)
+
+type DispatcherOption func(d *Dispatcher) error
+
 type Dispatcher struct {
-	client *http.Client
+	logger    *log.Logger
+	transport *http.Transport
+	client    *http.Client
+	rules     *netjail.Rules
```

Comment:
@jirevwe Also, `rules` should be behind the license gate.

Reply:
Got it